paywall/src/middlewares/web3Middleware.js

```diff
@@ -105,7 +105,16 @@ const web3Middleware = config => ({ getState, dispatch }) => {
   return function(next) {
     return function(action) {
       if (action.type === ADD_TRANSACTION) {
-        web3Service.getTransaction(action.transaction.hash)
+        if (action.transaction.input) {
```

Comment:
how would you feel about extracting this method into a library that handles this action? Its not pressing as the existing paradigm already exists inside our application but we should start to chop down the mega methods in our middleware

Reply:
I can see the value of this. On the other hand, the new design will pull all of this out (no more blockchain or locksmith inside paywall). Thus, this action won't be sent any longer. We will only receive updates to things like "locked" and the data will be all inside the key.

The new code is in `locksmithTransactions.js` (also modified in this PR), which doesn't do things quite the same way. What do you think?